astroid/brain/brain_builtin_inference.py

```diff
@@ -892,6 +892,26 @@ def _build_dict_with_elements(elements):
     return _build_dict_with_elements([])
 
 
+def _looks_like_copy_method(node):
+    func = node.func
+    return isinstance(func, nodes.Attribute) and func.attrname == "copy"
+
+
+def _has_copy_method(node):
+    return any(
+        isinstance(node, klass)
+        for klass in [nodes.Dict, nodes.List, nodes.Set, objects.FrozenSet]
+    )
+
+
+def _infer_copy_method(node, context=None):
+    inferred = list(node.func.expr.infer(context=context))
+    if all(_has_copy_method(x) for x in inferred):
```

Comment:
With the suggestions `_has_copy_method` is a single `isinstance` call. We could safe multiple function lookups / calls if we just inline it here.